File: Low Level Design (LLD)/Problems/05. FoodDeliverySystem/Java/payment/Payment.java
package payment;

import utils.PaymentStatus;

public class Payment {
    private String id;
    private String orderId;
    private double amount;
    private PaymentStatus status;

    public Payment(String id, String orderId, double amount, PaymentStatus status) {
        this.id = id;
        this.orderId = orderId;
        this.amount = amount;
        this.status = status;
    }

    public String getId() {
        return id;
    }

    public String getOrderId() {
        return orderId;
    }

    public double getAmount() {
        return amount;
    }

    public PaymentStatus getStatus() {
        return status;
    }
}

File: Low Level Design (LLD)/Problems/05. FoodDeliverySystem/Cpp/payment/src/Payment.cpp
#include "../include/Payment.h"

Payment::Payment() 
    : id(""), orderId(""), amount(0.0), status(PaymentStatus::PENDING) {}

Payment::Payment(const std::string& id, const std::string& orderId, double amount, PaymentStatus status)
    : id(id), orderId(orderId), amount(amount), status(status) {}

const std::string& Payment::getId() const {
    return id;
}

const std::string& Payment::getOrderId() const {
    return orderId;
}

double Payment::getAmount() const {
    return amount;
}

PaymentStatus Payment::getStatus() const {
    return status;
}

void Payment::setStatus(PaymentStatus newStatus) {
    status = newStatus;
}
